File: sdks/dotnet/src/Notilify/NotilifyException.cs
namespace Notilify;

public sealed class NotilifyException : Exception
{
    public int? StatusCode { get; }
    public object? DataPayload { get; }

    public NotilifyException(string message, int? statusCode = null, object? data = null, Exception? inner = null)
        : base(message, inner)
    {
        StatusCode = statusCode;
        DataPayload = data;
    }
}

File: sdks/php/src/NotilifyException.php
<?php

declare(strict_types=1);

namespace Notilify;

use RuntimeException;

final class NotilifyException extends RuntimeException
{
    public function __construct(
        string $message,
        public readonly ?int $statusCode = null,
        public readonly mixed $data = null,
    ) {
        parent::__construct($message);
    }
}
